wsi/drm/drmdisplay.cpp

```diff
@@ -425,19 +452,68 @@ float DrmDisplay::TransformGamma(float value, float gamma) const {
   return result;
 }
 
+void DrmDisplay::SetColorTransformMatrix(const float *color_transform_matrix,
+                                         int32_t color_transform_hint) const {
+  struct drm_color_ctm *ctm = (struct drm_color_ctm *)malloc(sizeof(struct drm_color_ctm));
+  if (!ctm) {
+    ETRACE("Cannot allocate CTM memory");
+    return;
+  }
+
+  switch (color_transform_hint) {
+    case 0: { // HAL_COLOR_TRANSFORM_IDENTITY
+      memset(ctm->matrix, 0, sizeof(ctm->matrix));
+      for (int i = 0; i < 3; i++) {
+        ctm->matrix[i * 3 + i] = (1ll << 32);
+      }
+      ApplyPendingCTM(ctm);
+      break;
+    }
+    case 1: { // HAL_COLOR_TRANSFORM_ARBITRARY_MATRIX
+      // Extract the coefficients from 4x4 Android HWC2 CTM to the drm 3x3 CTM.
```

Comment:
Probably define our own color hint enums and in iahwc2, translate them to our enum and pass it here, instead of case 1: 

Since our API also expects the same 4x4 matrix,  probably you can generalize this instead of saying "Android HWC2"